libnavigation-core/src/main/java/com/mapbox/navigation/core/SetRoutes.kt

```diff
@@ -37,6 +37,11 @@ internal sealed class SetRoutes {
         val legIndex: Int,
     ) : SetRoutes()
 
+    /**
+     * Triggered when routes list remain the same but **alternative route has been chosen as a primary**.
+     */
+    internal data class Reorder(val legIndex: Int) : SetRoutes()
```

Comment:
[NIT] As I understood it's an internal name which will be mapped to a different name for NN and SDK users. Maybe we can change it to be the same as comments above says:
```suggestion
    internal data class SwitchToAlternativeRoute(val legIndex: Int) : SetRoutes()
```
?

Reply:
After new discussion with NN team, this type is used when primary route changed and previous primary became alternative. As for me, both names `Reorder` and `SwitchToAlternativeRoute` can be used here, but the second option sounds very similar to `Alternatives`. What do you think?

Reply:
Reorder doesn't reflect the fact that primary route changed. You could have had [A, B, C] and then [A, C, B]. That sounds like a reorder to me, but it's not.
So IMO `SwitchToAlternative` suits better here.